storm-core/src/jvm/org/apache/storm/Config.java

```diff
@@ -1872,6 +1872,13 @@
     @isString
     public static final String TOPOLOGY_BOLTS_TUPLE_TIMESTAMP_FIELD_NAME = "topology.bolts.tuple.timestamp.field.name";
 
+    /**
+     * Bolt-specific configuration for windowed bolts to specify whether late tuples should be emitted on a stream
+     * called _late, instead of being logged with INFO level.
+     */
+    @isBoolean
+    public static final String TOPOLOGY_BOLTS_EMIT_LATE_TUPLE = "topology.bolts.emit.late.tuple";
```

Comment:
I realized that it was hard to achieve a consistent behavior in the earlier implementation, because at the moment of calling `declareOutputFields`, we only have access to the component specific parameters, but if the user puts the key in the global config but forgets to specify at component level, then the stream won't be declared, but in the `execute` method it will still going to try to emit messages to the non-existing stream.
On the other hand in the `initWindowManager` method, I can check whether the stream exist, and raise an exception if not. Does it seem okay to you? Because in this case I'd also prefer the per component based approach.